pkg/roundtripper/roundtripper.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package roundtripper
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+const (
+	service = "es"
+)
+
+type AWSSigningTransport struct {
+	t      http.RoundTripper
+	creds  aws.Credentials
+	region *string
+	log    log.Logger
+}
+
+func NewAWSSigningTransport(transport http.RoundTripper, region *string, log log.Logger) *AWSSigningTransport {
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(*region))
+	if err != nil {
+		_ = level.Error(log).Log("msg", "fail to load aws default config", "err", err)
+		os.Exit(1)
+	}
+
+	creds, err := cfg.Credentials.Retrieve(context.Background())
+	if err != nil {
+		_ = level.Error(log).Log("msg", "fail to retrive aws credentials", "err", err)
+		os.Exit(1)
```

Comment:
Same here. This function shouldn't call `os.Exit`

Reply:
Fixed